client/src/app/pages/applications/components/application-form/application-form.tsx

```diff
@@ -258,107 +253,36 @@ export const ApplicationForm: React.FC<ApplicationFormProps> = ({
     mode: "all",
   });
 
-  const buildBinaryFieldString = (
-    group: string,
-    artifact: string,
-    version: string,
-    packaging: string
-  ) => {
-    if (packaging) {
-      return `${group}:${artifact}:${version}:${packaging}`;
-    } else {
-      return `${group}:${artifact}:${version}`;
-    }
-  };
-
-  const onCreateApplicationSuccess = (response: AxiosResponse<Application>) => {
-    pushNotification({
-      title: t("toastr.success.createWhat", {
-        type: t("terms.application"),
-        what: response.data.name,
-      }),
-      variant: "success",
-    });
-    onClose();
-  };
-
-  const onUpdateApplicationSuccess = () => {
-    pushNotification({
-      title: t("toastr.success.save", {
-        type: t("terms.application"),
-      }),
-      variant: "success",
-    });
-    onClose();
-  };
-
-  const onCreateUpdateApplicationError = (error: AxiosError) => {
-    pushNotification({
-      title: getAxiosErrorMessage(error),
-      variant: "danger",
-    });
-  };
-
-  const { mutate: createApplication } = useCreateApplicationMutation(
-    onCreateApplicationSuccess,
-    onCreateUpdateApplicationError
-  );
-
-  const { mutate: updateApplication } = useUpdateApplicationMutation(
-    onUpdateApplicationSuccess,
-    onCreateUpdateApplicationError
-  );
-
   const onSubmit = (formValues: FormValues) => {
-    const matchingBusinessService = businessServices.find(
-      (businessService) =>
-        formValues?.businessServiceName === businessService.name
-    );
-
-    const matchingOwner = stakeholders.find(
-      (stakeholder) => formValues?.owner === stakeholder.name
-    );
-
-    const matchingContributors = stakeholders?.filter((stakeholder) =>
-      formValues.contributors.includes(stakeholder.name)
-    );
+    // Note: We need to manually retain the tags with source != "" in the payload
+    const tags = [...(tagsToRefs(formValues.tags) ?? []), ...nonManualTags];
 
     const payload: Application = {
+      id: formValues.id,
       name: formValues.name.trim(),
       description: formValues.description.trim(),
       comments: formValues.comments.trim(),
-      businessService: matchingBusinessService
+
+      businessService: businessServicesToRef(formValues.businessServiceName),
```

Comment:
Looks great - I found this util function naming to be slightly confusing since it is essentially looking to match one business service name to its matching ref. Would 
businessServiceToRef make more sense? It is probably a toss-up just wanted your thoughts.

Reply:
Yeah that would make more sense to go singular throughout